packages/controllers/src/snaps/endowments/cronjob.test.ts

```diff
@@ -0,0 +1,168 @@
+import { Caveat, PermissionType } from '@metamask/controllers';
+import { SnapCaveatType } from '@metamask/snap-utils';
+import {
+  getCronjobCaveatMapper,
+  cronjobEndowmentBuilder,
+  validateCronjobCaveat,
+  cronjobCaveatSpecifications,
+} from './cronjob';
+import { SnapEndowments } from '.';
+
+describe('endowment:cronjob', () => {
+  describe('specificationBuilder', () => {
+    it('builds the expected permission specification', () => {
+      const specification = cronjobEndowmentBuilder.specificationBuilder({});
+      expect(specification).toStrictEqual({
+        permissionType: PermissionType.Endowment,
+        targetKey: SnapEndowments.Cronjob,
+        endowmentGetter: expect.any(Function),
+        allowedCaveats: [SnapCaveatType.SnapCronjob],
+      });
+
+      expect(specification.endowmentGetter()).toBeUndefined();
+    });
+  });
+
+  describe('cronjobCaveatMapper', () => {
```

Comment:
Can you test that this works with the object format too?

Reply:
Added that format now.